core/src/main/java/com/yahoo/oak/BasicChunk.java

```diff
@@ -9,55 +9,503 @@
 import java.util.concurrent.atomic.AtomicInteger;
 import java.util.concurrent.atomic.AtomicReference;
 
+/*
+* EntryArray is array of entries based on primitive array of longs.
```

Comment:
It is weird that class BasicChunk starts with comment about EntryArray. At least BasicChunk should be explained in few words, saying that BasicChunk has an EntryArray, then about EntryArray.